### PR TITLE
[release-1.11] Bump CAPI to v1.5.5

### PR DESCRIPTION
**Pull request by mboersma**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:

Updates CAPI to [v1.5.5](https://github.com/kubernetes-sigs/cluster-api/releases/tag/v1.5.5) and all that entails. Also updates several tools in the `Makefile` since I noticed the versions had drifted between CAPZ and CAPI.

**Which issue(s) this PR fixes**:

N/A

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
Bump CAPI to v1.5.5
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -33,7 +33,7 @@ export GOPROXY
 export GO111MODULE=on
 
 # Kubebuilder.
-export KUBEBUILDER_ENVTEST_KUBERNETES_VERSION ?= 1.20.2
+export KUBEBUILDER_ENVTEST_KUBERNETES_VERSION ?= 1.28.0
```
Comment (nojnhuh) [inline Makefile:36]: I think the test failure is the same as #3420.

Comment (mboersma) [reply Makefile:36]: That's odd. The same change seems to have worked in #4464.

I'll revert this part and see if that gets things passing.

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm

File: test/e2e/config/azure-dev.yaml [hunk 2]
```diff
@@ -14,8 +14,8 @@ providers:
   - name: cluster-api
     type: CoreProvider
     versions:
-    - name: v1.5.4 # latest patch of earliest minor in supported v1beta1 releases; this is used for v1beta1 old --> v1beta1 latest clusterctl upgrades test only.
-      value: "https://github.com/kubernetes-sigs/cluster-api/releases/download/v1.5.4/core-components.yaml"
+    - name: v1.5.5 # latest patch of earliest minor in supported v1beta1 releases; this is used for v1beta1 old --> v1beta1 latest clusterctl upgrades test only.
```
Comment (nojnhuh) [inline test/e2e/config/azure-dev.yaml:17]: Should we also bump `images` above?

Comment (mboersma) [reply test/e2e/config/azure-dev.yaml:17]: Good catch! It was back on 1.5.3 so my simple find-and-replace missed it.

**Review by nojnhuh — COMMENTED** [review]
/lgtm
/approve

**Comment by mboersma** [comment]
/retest

Some netlify service error.

**Comment by nojnhuh** [comment]
@mboersma Might force-pushing without changing anything get the Netlify jobs to run again? Or maybe add an empty commit and squash it out right away?

**Pull request merged**